### PR TITLE
[JENKINS-49586] - Stop Serializing JDependParser to the disk (JEP-200 in 2.102+)

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-49586

- [x] Facelift the plugin
- [x] Add Jenkinsfile
- [x] Fix compatibility with Jenkins 2.102+

@reviewbybees @jglick

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Build the plugin using https://github.com/jenkins-infra/pipeline-library
+buildPlugin(jenkinsVersions: [null, '2.104'])
```
Comment (jglick) [inline Jenkinsfile]: 2.107 + https://github.com/jenkinsci/plugin-pom/pull/95

File: src/main/java/hudson/plugins/jdepend/JDependBuildAction.java [hunk 2]
```diff
@@ -8,23 +8,35 @@
...
+	private transient Run<?,?> owner;
+
+	public JDependBuildAction(JDependParser jDependParser) {
+		this(null, jDependParser);
+	}
+
+	/**
+	 * @deprecated Use {@link #JDependBuildAction(JDependParser)}
+	 */
+	@Deprecated
```
Comment (jglick) [inline src/main/java/hudson/plugins/jdepend/JDependBuildAction.java]: Probably pointless if you are deleting the `build` field anyway. Who would be calling this method from outside anyway?

File: src/main/java/hudson/plugins/jdepend/JDependBuildAction.java [hunk 3]
```diff
@@ -8,23 +8,35 @@
...
 /**
  * A build action to generate JDepend HTML reports
  * @author cflewis
  *
  */
-public class JDependBuildAction implements Action 
+public class JDependBuildAction implements RunAction2
 {
-	public final AbstractBuild<?, ?> build;
-	private final JDependParser jDependParser;
```
Comment (jglick) [inline src/main/java/hudson/plugins/jdepend/JDependBuildAction.java]: IIUC this is the actual fix—the plugin was for some reason saving an instance field for an object used only in the constructor? Sigh.

File: src/main/java/hudson/plugins/jdepend/JDependBuildAction.java (hunk 2 above)
Comment (oleg-nenashev) [reply src/main/java/hudson/plugins/jdepend/JDependBuildAction.java]: 🤷‍♂️  Prefer to keep binary compatibility when it does not cost too much

**Comment by oleg-nenashev** [comment]
@ndeloof could you please review/release that?

**Comment by oleg-nenashev** [comment]
@ndeloof OK, I have thought that formally you are a maintainer. Will request in the ML

**Comment by oleg-nenashev** [comment]
@ndeloof I discussed it with @daniel-beck, and we agreed that your approval is enough in this case. Could you please approve me taking ownership of this plugin for a while to spin the release? After some time I will mark it for adoption, the plan is similar to CppNCSS (https://groups.google.com/forum/#!msg/jenkinsci-dev/YU14gcHxJDg/CT5LpS1jBwAJ;context-place=forum/jenkinsci-dev)

**Comment by ndeloof** [comment]
@oleg-nenashev go for it

**Pull request merged**